src/dir.rs

```diff
@@ -225,7 +227,9 @@ impl Entry {
     /// See platform `readdir(3)` or `dirent(5)` manpage for when the file type is known;
     /// notably, some Linux filesystems don't implement this. The caller should use `stat` or
     /// `fstat` if this returns `None`.
+    #[cfg(not(any(target_os = "illumos", target_os = "solaris")))]
     pub fn file_type(&self) -> Option<Type> {
+        #[cfg(not(target_os = "illumos"))]
```

Comment:
I don't understand why the whole of `file_type()` was made conditional for illumos.  In my original patch [here](https://github.com/jclulow/nix/blob/illumos/src/dir.rs#L181-L202) it was not, because it seemed like it would then require less changes for consuming code -- we just return `None` in all cases and then the consumer knows (as per the documentation) to do their own `stat(2)`.  @jasonbking ?

Reply:
Oh my apologies, I missed the whole separate `file_type()` below.  This must be the result of merging our two porting effort.  Sorry about that!